src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
In Fuchsia's Rust stuff a lot of things are typed wrappers around handles, so for example if you have a handle to a channel to an IPC service speaking a protocol you can have a wrapper struct with Rust functions to send the relevant messages over the channel, and that wrapper will implement `HandleBase`.

Ideally we'll want to plumb the `EventedHandle` stuff up through Tokio, and expose a futures interface for the IPC system.

I haven't thought about what that would look like yet but I have a suspicion that it will be nicer with the generic on `EventedHandle`. On the other hand, given that most `HandleBase` structs are very thin wrappers, I wonder if there is a way to avoid the generic since it'll probably cause a lot of code bloat where it is instantiated almost identically for every wrapper.

I'll think about this and look at `tokio-core` and I'll try and come up with a definitive answer to how useful the generic is.